services/core/PlatformDriverAgent/HomeAssistantDriver.md

```diff
@@ -0,0 +1,239 @@
+
+```json
+{
+    "driver_config": {
+        "ip_address": "Your Home Assistant IP",
+        "access_token": "Your Home Assistant Access Token",
+        "volttron_topic": "your volttron topic",
+        "port": "Your Port"
+    },
+    "driver_type": "home_assistant",
+    "registry_config":"config://home_assistant_registers.json",
+    "interval": 30,
+    "timezone": "UTC"
+}
+```
+Your register file will can contain multiple entities but is best used with one. Using one keeps everything more organized.
+
+```json
+    {
+        "Point Name": "sensor.average_humidity_1621",
+        "Volttron Point Name": "sensor.average_humidity_1621",
+        "Units": "%",
+        "Units Details": "%",
+        "Writable": true,
+        "Starting Value": 20,
+        "Type": "float",
+        "Notes": "Average humidity of 1621",
+    },
+```
+
+Other devices such as thermostats will have multiple attributes, simply add the attributes in the attributes field.
+
+```json
+    {
+        "Point Name": "climate.thermostat",
+        "Volttron Point Name": "climate.thermostat",
+        "Units": "C",
+        "Units Details": "C",
+        "Writable": true,
+        "Starting Value": true,
+        "Type": "boolean",
+        "Notes": "lights bedroom",
+        "Attributes": {
+            "current_temperature": "current_temperature",
+            "current_humidity": "current_humidity"
+        }
+    },
+```
+
+
+Attributes can be found in developer tools or by opening the device in the GUI of Home Assistant. 
+
+![image](https://github.com/riley206/Rileys_volttron/assets/89715390/a367e61e-8b73-4f35-a179-dfda235ddcbe)
+
+
+We are now ready to get the driver running. Right now you can install the listener agent and a historian if you are looking to store data and view in VOLTTRON. Install the platform driver and make sure the home_assistant.py is in the interfaces folder. 
+
+Add the registers files and the config files into the VOLTTRON config store. 
+
+here are examples:  vctl config store platform.driver home_assistant_registers.csv HomeAssistant_Driver/home_assistant_registers.json --json
+                    vctl config store platform.driver devices/fakedevice examples/configurations/driver/fake.config
+
+Once this is complete you should be able to start the platform driver and tail the volttron log to see the devices being pulled in from Home Asssistant. 
+
+**VOLTTRON Home Assistant Driver - Code.**
+
+Pulling Data from Home Assistant:
+
+Pulling data from Home Assistant is managed by the get_entity_data function,. The get_entity_data function starts by creating the headers used for the REST API which includes your access token for authorization. We then create the URL with the last segment of the URL being the entitiy_id we have retireved through the registers file. This will grab the JSON data about that entity and return the response. 
```

Comment:
Data is retrieved from home assistant through the get_entity_data function.  It does the following:

1. Sets the headers
2. blah
3. blah

Reply:
Though this should be documented in the code itself.